MAINTAINERS.yml

```diff
@@ -766,6 +766,18 @@ Documentation:
   labels:
     - "area: I2S"
 
+"Drivers: I3C":
+  status: maintained
+  collaborators:
+    - dcpleung
```

Comment:
why not maintainer?

Reply:
I believe there is a formal process for one to become a maintainer...

Reply:
not in your case, you are already a maintainer and this is something new, so just put yourself as maintainer. Who else is better to be the maintainer from day one other than the one who introduced the area :)

Reply:
Updated.